### PR TITLE
[WIP][UI Tests]No Issue: Verify the page main menu items test

### PR DESCRIPTION
**Pull request by TejaswiKarasani**
This  UI Test written according to [this testrail](https://testrail.stage.mozaws.net/index.php?/cases/view/233849&group_by=cases:section_id&group_order=asc&group_id=31463) steps under Smoke & Sanity Test of High priority
### Pull Request checklist

- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### After merge
- [ ] **Milestone**: Make sure issues finished by this pull request are added to the [milestone](https://github.com/mozilla-mobile/fenix/milestones) of the version currently in development.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/androidTest/java/org/mozilla/fenix/ui/SmokeTest.kt [hunk 1]
```diff
@@ -62,4 +64,414 @@ class SmokeTest {
...
+            activityTestRule.getActivity()
+                .setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE)
+        }.openThreeDotMenu {
+            // verifyThreeDotMainMenuItems()
+        }.openHistory {
+            // verifyTestPageUrl(defaultWebPage.url)
+        }.goBackToBrowser {
+        }.openThreeDotMenu {
+            activityTestRule.getActivity()
+                .setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE)
```
Comment (isabelrios) [inline app/src/androidTest/java/org/mozilla/fenix/ui/SmokeTest.kt:233]: Do you need this everytime? I think if you set that at the very beginning then the whole test runs on landscape, at least that's what I saw when trying that out

Comment (TejaswiKarasani) [reply app/src/androidTest/java/org/mozilla/fenix/ui/SmokeTest.kt:233]: Okay

File: app/src/androidTest/java/org/mozilla/fenix/ui/SmokeTest.kt [hunk 2]
```diff
@@ -73,12 +73,11 @@ class SmokeTest {
         }.enterURLAndEnterToBrowser(defaultWebPage.url) {
         }.openThreeDotMenu {
             verifyThreeDotMainMenuItems()
-            // addOns()
-            /*}.clickAddOnsReportSiteIssue {
+        }.clickAddOnsReportSiteIssue {
             verifyUrl("https://webcompat.com/issues/new")
```
Comment (isabelrios) [inline app/src/androidTest/java/org/mozilla/fenix/ui/SmokeTest.kt:77]: To have it working for me, I had to remove the "https://" part

File: app/src/androidTest/java/org/mozilla/fenix/ui/SmokeTest.kt [hunk 3]
```diff
@@ -113,7 +112,7 @@ class SmokeTest {
             verifyShortcutNameField(defaultWebPage.title)
             clickAddShortcutButton()
             clickAddAutomaticallyButton()
-            // verifyShortcutIcon()
+            verifyShortcutIcon()
```
Comment (isabelrios) [inline app/src/androidTest/java/org/mozilla/fenix/ui/SmokeTest.kt:115]: I meant removing this line, not sure if uncommenting would work

File: app/src/androidTest/java/org/mozilla/fenix/ui/SmokeTest.kt [hunk 4]
```diff
@@ -152,6 +151,11 @@ class SmokeTest {
                 }.enterURLAndEnterToBrowser(defaultWebPage.url) {
                 }.openThreeDotMenu {
                     verifyThreeDotMainMenuItems()
+                }.clickAddOnsReportSiteIssue {
+                    verifyUrl("https://webcompat.com/issues/new")
```
Comment (isabelrios) [inline app/src/androidTest/java/org/mozilla/fenix/ui/SmokeTest.kt:155]: Same here with the http...

File: app/src/androidTest/java/org/mozilla/fenix/ui/robots/ThreeDotMenuMainRobot.kt [hunk 5]
```diff
@@ -365,7 +364,7 @@ private fun assertSettingsButton() = settingsButton()
 private fun addOnsButton() = onView(allOf(withText("Add-ons")))
 private fun assertAddOnsButton() = addOnsButton()
     .check(matches(withEffectiveVisibility(Visibility.VISIBLE)))
-private fun addOnsReportSiteIssueButton() = onView(allOf(withText("Report Site Issue...")))
+private fun addOnsReportSiteIssueButton() = onView(allOf(withText("Report Site Issue…")))
```
Comment (isabelrios) [inline app/src/androidTest/java/org/mozilla/fenix/ui/robots/ThreeDotMenuMainRobot.kt:367]: 👍

File: app/src/androidTest/java/org/mozilla/fenix/ui/SmokeTest.kt (hunk 2 above)
Comment (TejaswiKarasani) [reply app/src/androidTest/java/org/mozilla/fenix/ui/SmokeTest.kt:77]: okay

File: app/src/androidTest/java/org/mozilla/fenix/ui/SmokeTest.kt (hunk 3 above)
Comment (TejaswiKarasani) [reply app/src/androidTest/java/org/mozilla/fenix/ui/SmokeTest.kt:115]: okay

File: app/src/androidTest/java/org/mozilla/fenix/ui/SmokeTest.kt (hunk 4 above)
Comment (TejaswiKarasani) [reply app/src/androidTest/java/org/mozilla/fenix/ui/SmokeTest.kt:155]: okay

File: app/src/androidTest/java/org/mozilla/fenix/ui/robots/ThreeDotMenuMainRobot.kt (hunk 5 above)
Comment (TejaswiKarasani) [reply app/src/androidTest/java/org/mozilla/fenix/ui/robots/ThreeDotMenuMainRobot.kt:367]: :)

File: app/src/androidTest/java/org/mozilla/fenix/ui/SmokeTest.kt [hunk 6]
```diff
@@ -62,4 +63,160 @@ class SmokeTest {
...
+    @Test
+    fun verifyPageMainMenuItemsListInPortraitPrivateModeTest() {
+        val defaultWebPage = TestAssetHelper.getGenericAsset(mockWebServer, 1)
+        // Add this to check openInApp and also youtube is a default app available in every Android emulator/device
+        val youtubeUrl = "www.youtube.com"
+
+        homeScreen {
+            togglePrivateBrowsingModeOnOff()
+            navigationToolbar {
+                navigationToolbar {
```
Comment (isabelrios) [inline app/src/androidTest/java/org/mozilla/fenix/ui/SmokeTest.kt:153]: Just realized about this...This navigationToolbar is duplicated, this or the above can be removed

Comment (TejaswiKarasani) [reply app/src/androidTest/java/org/mozilla/fenix/ui/SmokeTest.kt:211]: Yes have to remove that. It was a mistake over there

File: app/src/androidTest/java/org/mozilla/fenix/ui/SmokeTest.kt [hunk 7]
```diff
@@ -62,4 +63,158 @@ class SmokeTest {
...
+            }.openThreeDotMenu {
+                verifyOpenInAppButton()
+            }
+        }
+    }
+
+    @Test
+    fun verifyPageMainMenuItemsListInPortraitPrivateModeTest() {
+        val defaultWebPage = TestAssetHelper.getGenericAsset(mockWebServer, 1)
+        // Add this to check openInApp and also youtube is a default app available in every Android emulator/device
```
[the remark this answers is not in the document]
Comment (TejaswiKarasani) [reply app/src/androidTest/java/org/mozilla/fenix/ui/SmokeTest.kt:147]: okay

**Pull request merged**